cmd/ebpf/xdp/utils/parsers.h

```diff
@@ -77,6 +81,30 @@ static __always_inline int parse_udp(struct packet_context *ctx) {
     return bpf_ntohs(udp->dest);
 }
 
+static __always_inline int parse_tcp(struct packet_context *ctx) {
+    struct tcphdr *tcp = (struct tcphdr *)ctx->data;
+    if ((const char *)(tcp + 1) > ctx->data_end)
+        return -1;
+
+    //TODO: parse header lenght correctly (tcp options)
+
+    ctx->data += sizeof(*tcp);
+    ctx->tcp = tcp;
+    return bpf_ntohs(tcp->dest);
+}
+
+static __always_inline int parse_l4(int ip_protocol, struct packet_context *ctx)
+{
+    switch (ip_protocol) {
+        case IPPROTO_UDP:
+            return parse_udp(ctx);
+        case IPPROTO_TCP:
+            return parse_tcp(ctx);
+        default:
+            return 0;
```

Comment:
Shouldn't the default return code be -1? I mean, in lines 213 and 237 (calling function parse_l4) an error code is considered as -1, and 0 code is never checked.

Reply:
Non-negative return value means success in this case. Not so perfect idea. Caused by `parse_udp` and `parse_tcp` signatures which both returns port values.